test/compatibility-test/application/application_controller_test.go

```diff
@@ -0,0 +1,1555 @@
+/*
+Copyright 2020 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package application
```

Comment:
why copy so many lines of code here from application controller?

Reply:
The purpose of compatibility-test is to test application/ac controller can work well with **cluster scope** workload/trait definition. So i put old crd yaml file in /test/compatibility-test/testdata and copy related application and ac suit_test code here.